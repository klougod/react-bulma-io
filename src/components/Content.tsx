import React from 'react'
import PropTypes from 'prop-types'

export const Content = ({ className, children }: any) => (
  <div className={`content ${className}`}>{children}</div>
)

Content.defaultProps = {
  className: ''
}

Content.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default Content
