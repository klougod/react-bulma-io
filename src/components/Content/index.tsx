import React, { ReactNode } from 'react'

interface ContentProps {
  className?: string
  children?: ReactNode
}
export const Content = ({ className, children }: ContentProps) => (
  <div className={`content ${className}`}>{children}</div>
)

Content.defaultProps = {
  className: ''
}

export default Content
