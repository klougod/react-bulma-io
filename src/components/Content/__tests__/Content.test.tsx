import React from 'react'
import Enzyme from 'enzyme'

import Content from '../'

describe('Content component test', function () {
  it('should render without crash without props', function () {
    const contentMock = Enzyme.shallow(<Content />)
    expect(contentMock).toMatchSnapshot()
  })

  it('should render without crash with props', function () {
    const contentMockProps = {
      className: 'contentCLass',
    }
    const contentMock = Enzyme.shallow(<Content {...contentMockProps} />)
    expect(contentMock).toMatchSnapshot()
  })
})
