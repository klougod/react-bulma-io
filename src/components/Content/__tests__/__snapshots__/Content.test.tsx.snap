// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content component test should render without crash with props 1`] = `
<div
  className="content contentCLass"
/>
`;

exports[`Content component test should render without crash without props 1`] = `
<div
  className="content "
/>
`;
